### PR TITLE
DEV: add support for more tree distance metrics

### PR DESCRIPTION
**Pull request by rmcar17**
Addresses #1644 and #1590.

Also fixes issues in the cookbook documentation for `simple_trees.rst` where examples that used `make_tree` imported `load_tree` instead.

### CODE REVIEW

File: doc/cookbook/simple_trees.rst [hunk 1]
```diff
@@ -437,12 +437,61 @@ Branch lengths don't matter.
...
 
-    from cogent3 import load_tree
+    from cogent3 import make_tree
 
     tr1 = make_tree("(B:0.2,(C:0.2,D:0.2)F:0.2)G;")
     tr2 = make_tree("((C:0.1,D:0.1)F:0.1,B:0.1)G;")
     tr1.same_topology(tr2)
 
+Measure topological distances between two trees
+"""""""""""""""""""""""""""""""""""
```
Comment (GavinHuttley) [inline doc/cookbook/simple_trees.rst:447]: For rst, the number of heading characters (the `"` in this case) must be ≥ the length of the line above

File: src/cogent3/core/tree.py [hunk 2]
```diff
@@ -1598,7 +1599,74 @@ def rooted_with_tip(self, outgroup_name):
...
+            Default is "matching".
+
+        Returns
+        -------
+        int
+            the chosen distance between the two trees.
+
+        Notes
+        -----
+        The Lin-Rajan-Moret distance and Matching Cluster distance displays
```
Comment (GavinHuttley) [inline src/cogent3/core/tree.py:1636]: change "displays" to "display"

File: src/cogent3/core/tree.py [hunk 3]
```diff
@@ -1598,7 +1599,74 @@ def rooted_with_tip(self, outgroup_name):
...
+        superior statistical properties than the Robinson-Foulds distance on
+        unrooted and rooted trees respectively.
+
+        see: Bogdanowicz, D., & Giaro, K. (2013).
```
Comment (GavinHuttley) [inline src/cogent3/core/tree.py:1640]: For including citations, see the [numpydoc documentation](https://numpydoc.readthedocs.io/en/latest/format.html#references).

File: src/cogent3/phylo/tree_distance.py [hunk 4]
```diff
@@ -1,29 +1,89 @@
 from itertools import product
-from typing import List
+from typing import TYPE_CHECKING
```
Comment (GavinHuttley) [inline src/cogent3/phylo/tree_distance.py:2]: Nice!

File: src/cogent3/phylo/tree_distance.py [hunk 5]
```diff
@@ -1,29 +1,89 @@
...
     Returns
     -------
-    float
+    int
         the Lin-Rajan-Moret distance
 
     Notes
     -----
-    see: Lin et al. 2012 A Metric for Phylogenetic Trees Based on Matching
+    see: Lin et al. 2012
```
Comment (GavinHuttley) [inline src/cogent3/phylo/tree_distance.py:85]: we should also use the correct reference formatting here too

File: src/cogent3/phylo/tree_distance.py [hunk 6]
```diff
@@ -41,16 +101,97 @@ def lin_rajan_moret(tree1: "PhyloNode", tree2: "PhyloNode") -> float:
...
     matching_distance = _matched_distance(vector1, vector2)
 
-    return float(matching_distance)
+    return matching_distance
+
+
+def rooted_robinson_foulds(tree1: "TreeNode", tree2: "TreeNode") -> int:
+    """calculate the robinson-foulds distance between two rooted trees.
+
+    for rooted trees, the robinson-foulds distance is defined as the
```
Comment (GavinHuttley) [inline src/cogent3/phylo/tree_distance.py:110]: This text should be under a Notes section and given proper English grammar formatting (sentences start with capital letter, etc..)

File: src/cogent3/phylo/tree_distance.py [hunk 7]
```diff
@@ -41,16 +101,97 @@ def lin_rajan_moret(tree1: "PhyloNode", tree2: "PhyloNode") -> float:
...
+        trees to calculate distance between
+
+    Returns
+    -------
+    int
+        the matching cluster distance
+
+    Notes
+    -----
+    see: Bogdanowicz, D., & Giaro, K. (2013).
```
Comment (GavinHuttley) [inline src/cogent3/phylo/tree_distance.py:154]: citation format fixes

File: tests/test_tree_distance.py [hunk 8]
```diff
@@ -3,17 +3,86 @@
...
 from cogent3 import make_tree
-from cogent3.phylo.tree_distance import lin_rajan_moret
+from cogent3.phylo.tree_distance import (
+    lin_rajan_moret,
+    matching_cluster_distance,
+    rooted_robinson_foulds,
+    unrooted_robinson_foulds,
+)
+
+ROOTED_DISTANCE_MEASURES = (matching_cluster_distance, rooted_robinson_foulds)
```
Comment (GavinHuttley) [inline tests/test_tree_distance.py:13]: remove surrounding parentheses, they're not required

File: tests/test_tree_distance.py [hunk 9]
```diff
@@ -3,17 +3,86 @@
...
+UNROOTED_DISTANCE_MEASURES = (lin_rajan_moret, unrooted_robinson_foulds)
+
+
+# unrooted rf distance
+def test_unrooted_rf_different_trees():
+    a = make_tree(treestring="(a, b, (c, (d, e)));")
+
+    b = make_tree(treestring="(a, b, (d, (c, e)));")
```
Comment (GavinHuttley) [inline tests/test_tree_distance.py:21]: add a comment about where the expected values come from, i.e. what makes them "expected"?

File: tests/test_tree_distance.py [hunk 10]
```diff
@@ -3,17 +3,86 @@
...
+    a = make_tree(treestring="(a, (b, (c, d)), (e, (f, (g, h))));")
+    b = make_tree(treestring="(a, (c, (b, d)), (h, (f, (g, e))));")
+    distance = unrooted_robinson_foulds(a, b)
+    assert distance == 6
+    distance = unrooted_robinson_foulds(b, a)
+    assert distance == 6
+
+
+def test_unrooted_rf_same_tree():
+    a = make_tree(treestring="(a, b, (c, d));")
```
Comment (GavinHuttley) [inline tests/test_tree_distance.py:48]: can you add a comment that by same tree you mean the same topology, but a different "orientation"

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
Nice code! Most of the changes concern slight tweaks to restructured text formatting. In looking over your tests, I think we should be thinking about communicating to future developers. For instance, why is a value expected?
Another question is, are their cases where thew different tree metrics are expected to be identical and cases where they are expected to be different. Have you covered these?

File: doc/cookbook/simple_trees.rst (hunk 1 above)
Comment (rmcar17) [reply doc/cookbook/simple_trees.rst]: Resolved in 3813866

File: src/cogent3/core/tree.py (hunk 2 above)
Comment (rmcar17) [reply src/cogent3/core/tree.py]: Done between f7929a9 / d705da2

**Review by GavinHuttley — APPROVED** [review]
Thanks @rmcar17 🎉 !

**Comment by rmcar17** [comment]
Cheers!

**Pull request merged**